### PR TITLE
Add exponential backoff to reading config file

### PR DESCRIPTION
**Pull request by aaronburtle**
## Why make this change?

Closes https://github.com/Azure/data-api-builder/issues/2119

## What is this change?

We add a try catch around the file reading in the `FileSystemRuntimeConfigLoader` class, and implement an exponential backoff for reading said file.

## How was this tested?

Manually tested and ran against the test suite. A manual test can be run by having an editor with the file in question open and then closing the editor during the reading process.

## Sample Request(s)

N/A

### CODE REVIEW

File: src/Config/Utilities/FileUtilities.cs [hunk 1]
```diff
@@ -42,7 +42,7 @@ public static byte[] ComputeHash(IFileSystem fileSystem, string filePath)
         int runCount = 1;
 
         // Maximum 2^_runLimit seconds of wait time due to retries.
```
Comment (RubenCerna2079) [inline src/Config/Utilities/FileUtilities.cs:44]: Change the variable name in the comment
```suggestion
        // Maximum 2^RunLimit seconds of wait time due to retries.
```

**Comment by RubenCerna2079** [comment]
LGTM

**Review by abhishekkumams — APPROVED** [review]

**Pull request merged**